x-pack/solutions/observability/plugins/slo/public/components/burn_rate_rule_editor/slo_selector.tsx

```diff
@@ -23,8 +24,11 @@ function SloSelector({ initialSlo, onSelected, errors }: Props) {
   const [options, setOptions] = useState<Array<EuiComboBoxOptionOption<string>>>([]);
   const [selectedOptions, setSelectedOptions] = useState<Array<EuiComboBoxOptionOption<string>>>();
   const [searchValue, setSearchValue] = useState<string>('');
-  const { isLoading, data } = useFetchSloDefinitions({ name: searchValue });
+  const { isLoading, isInitialLoading, data, hasLoadedOnlyOnce } = useFetchSloDefinitions({
+    name: searchValue,
+  });
   const hasError = errors !== undefined && errors.length > 0;
+  const hasNoSlos = data?.results.length === 0;
```

Comment:
if data is undefined, you'll get hasNoSlos = false. I think you can use `isInitialLoading` and `isLoading` (the ones from react query hook, not your custom implementation) and you should be fine with these conditions:

```
const hasNoSlos = !data || data.total === 0
if (isInitialLoading) { render spinner }
if (!isLoading && hasNoSlo) { return emptyStat }
render selector
```